src/main/java/scrolls/elder/model/ModelManager.java

```diff
@@ -147,7 +147,25 @@ public void updateFilteredPersonList(Predicate<Person> predicate) {
         requireNonNull(predicate);
         filteredPersons.setPredicate(predicate);
         filteredVolunteers.setPredicate(person -> predicate.test(person) && person.isVolunteer());
-        filteredBefriendees.setPredicate(person -> predicate.test(person) && !(person.isVolunteer()));
+        filteredBefriendees.setPredicate(person -> predicate.test(person) && person.isBefriendee());
+    }
+
+    @Override
+    public void updateFilteredVolunteerList(Predicate<Person> predicate) {
+        requireNonNull(predicate);
+
+        // Only apply predicate to volunteers
+        filteredPersons.setPredicate(person -> person.isBefriendee() || predicate.test(person));
+        filteredVolunteers.setPredicate(person -> predicate.test(person) && person.isVolunteer());
+    }
+
+    @Override
+    public void updateFilteredBefriendeeList(Predicate<Person> predicate) {
+        requireNonNull(predicate);
+
+        // Only apply predicate to befriendees
+        filteredPersons.setPredicate(person -> predicate.test(person) || person.isVolunteer());
+        filteredBefriendees.setPredicate(person -> predicate.test(person) && person.isBefriendee());
```

Comment:
May have to move this functionality down into the Datastore structure

Reply:
Done! Thank you for the insight!